clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -37,6 +41,11 @@
 
     private static final Logger log = LoggerFactory.getLogger(Metadata.class);
 
+    /* Producer sets topic expiry to this value when a message is available to send. Expiry is updated on next metadata refresh */
+    private static final Long TOPIC_EXPIRY_NEEDS_UPDATE = 0L;
+    /* Consumer coordinator does not need automated expiry since topics are managed independently by the coordinator */
+    private static final Long TOPIC_EXPIRY_NONE = Long.MAX_VALUE;
```

Comment:
This might be better named `TOPIC_EXPIRY_NEVER` since it indicates it should never expire, not that you have no info about expiration time yet.